web/src/utils/factory-management/validation.ts

```diff
@@ -38,9 +39,34 @@ export const validateFactories = (factories: Factory[]) => {
         delete factory.dependencies.requests[depFacId]
       }
     })
+
+    // Check for invalid products and remove them from factories
+    // For instance if somehow a product has an amount of 0, which should not be possible, remove the product and recalculate.
+    factory.products.forEach((product, productIndex) => {
+      if (product === null) {
+        console.error(`VALIDATION ERROR: Factory "${factory.name}" (${factory.id}) has a product that is somehow null. Removing the product.`)
+        factory.products.splice(productIndex, 1)
+      }
+
+      if (product && product.amount <= 0) {
+        hasErrors = true
+        console.error(`VALIDATION ERROR: Factory "${factory.name}" (${factory.id}) has a product with an amount of 0 or less. Setting to 1.`)
+
+        product.amount = 1
+      }
+
+      if (factory.products.length === 0) {
+        factory.products = [] // Ensure there's no lurking bad data
+      }
+
+      // Recalculate right now
+      calculateFactory(factory, factories, gameData)
```

Comment:
Need to make do the calculation if only the above conditions are touched, otherwise this will calculate every time.